pscript/parser2.py

```diff
@@ -934,17 +934,17 @@ def parse_FunctionDef(self, node, lambda_=False, asyn=False):
             if not node.kwargs_node:
                 code.append(", '%s'" % func_name or 'anonymous')
             code.append(');')
+            if vararg_code2:
+                code.append(self.lf(vararg_code2))
             # Apply values of positional args
             # inside if, because standard arguments are invalid
             args_var = 'arguments[0].flx_args'
             if len(argnames) > 1:
                 args_var = self.dummy('args')
                 code.append(self.lf('%s = arguments[0].flx_args;' % args_var))
-            for i, name in enumerate(argnames):
+            for i, name in reversed(list(enumerate(argnames))):
                 code.append(self.lf('%s = %s[%i];' % (name, args_var, i)))
```

Comment:
I'm afraid that I don't follow. As far as I can see, the order should not matter, since we do not change `arguments[0].flx_args`. What am I missing?

Could you add a test for the problem you were having? That would help in understanding this change, and make sure that it's not accidentally broken at a later time :)

Reply:
@almarklein whe you call a function like this one `def partial2(a1, a2, *args, **kwargs):` your "real" arguments dictionary is in `a1` (which is the same as `arguments[0]`).

The following happens (at least in Chrome):

1. You extract **kwargs from arguments[0]
2. You extract and assign positional arguments starting with the 1st one (`a1`)
3. So you've just cleared original `arguments[0]` (`a1`) by assigning a new value to `a1`
4. You cannot access original `arguments[0]` anymore.